[quant] quantized::conv_transpose2d_prepack

z-a-f opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #37761 [quant] quantized.ConvTranspose1d and quantized.ConvTranspose2d
* #37759 [quant] quantized::conv1d_prepack, quantized::conv_transpose1d_prepack, and quantized::conv_transpose1d
* #37711 [quant] quantized::conv_transpose2d
* **#37777 [quant] quantized::conv_transpose2d_prepack**
* #37497 [quant] QNNPACK factoring out common structs from conv/deconv
* #36790 [quant] QNNPACK deconv kernel and tests
* #37405 [quant] QNNPACK deconvolution packing

Differential Revision: [D21386462](https://our.internmc.facebook.com/intern/diff/D21386462)

Commit:
[quant] quantized::conv_transpose2d_prepack

[ghstack-poisoned]

supriyar commented on aten/src/ATen/native/quantized/cpu/qconv_prepack.cpp:
```diff
@@ -238,8 +281,8 @@ class QConvPackWeightInt8 final {
     // but PyTorch lays them out as {out_c, in_c/groups, kH, kW}
-    const size_t out_ch = weight.size(0);
-    const size_t in_ch = weight.size(1) * groups;
+    const size_t out_ch = transpose ? weight.size(1) * groups : weight.size(0);
```
Could you update the comment above for conv_transpose as well?

supriyar commented on aten/src/ATen/native/quantized/cpu/qconv_prepack.cpp:
```diff
@@ -227,7 +266,11 @@ class QConvPackWeightInt8 final {
     TORCH_CHECK(
-        padding.size() == 2,
+        input_padding.size() == 2,
+        "quantized::conv2d_prepack (qnnpack): Specify top/left padding only. "
```
Can we update the error message to specify input/output padding (here and below)?

supriyar reviewed (commented):
Should we also have a conv_transpose2d_unpack function? It will be the same as conv2d_unpack but might be better to have a diff function to be consistent with prepack

Commit:
Update on "[quant] quantized::conv_transpose2d_prepack"


Differential Revision: [D21386462](https://our.internmc.facebook.com/intern/diff/D21386462)

[ghstack-poisoned]
